assemblyline/values.yaml

```
@@ -132,6 +132,17 @@ ingressHost:
 # If a tls secret name is given here, that certificate will be used for https
 # If left null, a self signed cert will be generated.
 tlsSecretName:
+useSharedTLSCert:
+
+# Use this to pull TLS from KeyVault using CSI driver
+secretsProvider:
+  enabled: true
+  className: "azure-kvname-secrets-provider" # Replace 'kvname' with the actual KeyVault name
+  tlsSecretName: "tls-secret-for-app" # Descriptive name for the secret in your Kubernetes cluster
+  tlsObjectName: "app-tls-certificate" # Name of the TLS certificate object in Azure KeyVault
+  identity: "keyvault-identity" # Azure identity to access KeyVault (usually a Managed Identity)
+  tenantId: "azure-tenant-id" # Your Azure Tenant ID
+  keyvault: "myapp-keyvault" # Name of the Azure KeyVault to use
```

Perhaps to generalize for other providers like HashiCorp, AWS, and GCP?

```suggestion
secretProvider:
  className: # Name of SecretProvider class object
  tlsSecretName: "tls-secret-for-app" # Descriptive name for the secret in your Kubernetes cluster
  tlsObjectName: "app-tls-certificate" # Name of the TLS certificate object in Azure KeyVault
  provider: # Name of your provider
  parameters: {} # Used to configure your CSI relative to the provider
```
```